Platforms/QemuQ35Pkg/Tcg/Tcg2ProtocolStub/Tcg2ProtocolStub.c

```diff
@@ -0,0 +1,45 @@
+/** @file
+  This module installs the gEfiTcg2ProtocolGuid to satisfy
+  DEPEX expressions in modules which do not actually use the
+  protocol.
+
+  Copyright (C) Microsoft Corporation. All rights reserved.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#include <Library/DebugLib.h>
+#include <Library/UefiBootServicesTableLib.h>
+
+/**
+  Install NULL gEfiTcg2ProtocolGuid
+
+  @param[in] ImageHandle  The firmware allocated handle for the EFI image.
+  @param[in] SystemTable  A pointer to the EFI System Table.
+
+  @retval EFI_SUCCESS     The entry point is executed successfully.
+
+**/
+EFI_STATUS
+EFIAPI
+Tcg2ProtocolStubEntryPoint (
+  IN    EFI_HANDLE        ImageHandle,
+  IN    EFI_SYSTEM_TABLE  *SystemTable
+  )
+{
+  EFI_STATUS  Status;
+
+  Status = gBS->InstallProtocolInterface (
+                  &ImageHandle,
+                  &gEfiTcg2ProtocolGuid,
+                  EFI_NATIVE_INTERFACE,
+                  NULL

```

Comment:
Have you checked that this will work? Some of the code that I checked (standard code in repos) assumes that if Tcg2Protocol returns success, that the protocol will not be null

Reply:
Yes this works. The MOR lock drivers depend on the protocol but do not use it. This driver is locked behind TPM_ENABLE==FALSE (if TPM_ENABLE==TRUE then the Tcg2Dxe driver is included).